### ISSUE TITLE
How to write comments: discuss

### ISSUE THREAD

**Issue by notwaldorf** [GoogleChrome/dash-elements#6]
Right now there are two elements and two styles of writing comments:
- say what the function does wrt to the spec (from Surma): [constructor](https://github.com/GoogleChrome/dash-elements/blob/master/elements/tab-panel/tab-panel.js#L25), [createdCallback](https://github.com/GoogleChrome/dash-elements/blob/master/elements/tab-panel/tab-panel.js#L37)
- say what the element does in that function (from Rob): [constructor](https://github.com/GoogleChrome/dash-elements/pull/2/files#diff-6382a28e520927272dfc5cc0a80ca444R22), [createdCallback](https://github.com/GoogleChrome/dash-elements/pull/2/files#diff-6382a28e520927272dfc5cc0a80ca444R30)

Both are great! Should we do both?

**Comment by surma** [comment]
I actually didn’t follow my own review nit I gave to Rob: The block comment above a function should give a high-level description of what that function does. I’ll adjust my comments.

That doesn’t mean we should not mention the spec, but I’d be interested to hear what y’all think about that. It might get repetitive across elements, but it’s also an important thing to teach with regards to CEv1. WDYT?

**Comment by robdodson** [comment]
We could make a kind of generic element <dash-example> that explains the features of CE and have folks lead off with that maybe?

**Comment by robdodson** [comment]
```
class DashExample extends HTMLElement {
  // Explain how constructor works
  constructor

  // Explain how connected works
  connectedCallback

  // Explain how attribute changed works
  attributeChangedCallback
}
```

**Comment by surma** [comment]
Yeah could do. Doesn’t feel right to me, but I also don’t have a better suggestion except for duplicating it across all elements.

**Comment by robdodson** [comment]
Another option would be to write a kind of intro/preamble to the project
linking to Eric's primer on custom elements
https://developers.google.com/web/fundamentals/getting-started/primers/customelements

**Comment by surma** [comment]
What do y’all think is the main use-case for external folks of this?

Are they going to read all of DASH because they are interested or are they on the hunt for an accordion and find this and just gonna read through that one element?

My gut says it’s the latter, and that would speak in favor duplicating that information across multiple elements. I do realize that it’s hard to draw the line as to what is assumed to be “common knowledge” and what needs to be explained, so having a collection of links in the intro is a really good idea imo.

**Comment by robdodson** [comment]
The latter. Yeah maybe just some primer links in the beginning of each
element would suffice

**Comment by surma** [comment]
Okay, let’s add a section to [the template](https://github.com/GoogleChrome/dash-elements/blob/master/site-resources/element.tpl.html#L20-L22) :) 

Feel free to grab this issue, otherwise I’ll do it at some point.

**Comment by surma** [comment]
So bottom line: We will go with @robdodson’s version and move the spec relevant stuff to the intro section. Agreed?

**Comment by robdodson** [comment]
agreed. btw sorry for lagging i've been traveling the last 4 days

**Comment by surma** [comment]
Fixed with #15
